tests/PHPStan/Analyser/data/falsy-isset.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace FalsyIsset;
+
+use function PHPStan\Testing\assertType;
+use function PHPStan\Testing\assertVariableCertainty;
+use PHPStan\TrinaryLogic;
+
+class HelloWorld
+{
+	public function undefinedVar(): void
+	{
+		if (isset($a['bar'])) {
+			assertType("*ERROR*", $a);
+		} else {
+			assertType("*ERROR*", $a);
+		}
+	}
+
+	public function maybeCertainNull(): void
+	{
+		if (rand() % 2) {
+			$a = ['bar' => null];
+			if (rand() % 3) {
+				$a = ['bar' => 'hello'];
+			}
+		}
+		assertVariableCertainty(TrinaryLogic::createMaybe(), $a);
+
+		assertType("array{bar?: null}|array{bar?: 'hello'}", $a);
+		if (isset($a['bar'])) {
+			assertVariableCertainty(TrinaryLogic::createYes(), $a);
+			assertType("array{bar: 'hello'}", $a);
+			$a['bar'] = 1;
+			assertType("array{bar: 1}", $a);
+		} else {
+			assertVariableCertainty(TrinaryLogic::createMaybe(), $a);
+			assertType('array{bar?: null}', $a);
+		}
+
+		assertVariableCertainty(TrinaryLogic::createMaybe(), $a);
+		assertType("array{bar?: null}|array{bar: 1}", $a);
+	}
+
+	public function maybeCertainNonNull(): void
+	{
+		if (rand() % 2) {
+			$a = ['bar' => 'world'];
+			if (rand() % 3) {
+				$a = ['bar' => 'hello'];
+			}
+		}
+		assertVariableCertainty(TrinaryLogic::createMaybe(), $a);
+
+		assertType("array{bar?: 'hello'}|array{bar?: 'world'}", $a);
+		if (isset($a['bar'])) {
+			assertVariableCertainty(TrinaryLogic::createYes(), $a);
+			assertType("array{bar: 'hello'}|array{bar: 'world'}", $a);
+			$a['bar'] = 1;
+			assertType("array{bar: 1}", $a);
+		} else {
+			assertVariableCertainty(TrinaryLogic::createNo(), $a);
+			assertType('*ERROR*', $a);
```

Comment:
as requested, I am using certainty=no for the non-nullable & !isset -case.

this also turns the type after the if/else into a `*ERROR* - I think because the type after if/else is determined by union of if- and else- branch, which because of union with NO

Reply:
hmm maybe the NotIssetExpr should be handled like an early return, so it does not influence the type after the if/else?